src/Persistent/Api.hs

```diff
@@ -126,11 +129,52 @@ makeLenses ''AulaData
 
 deriveSafeCopy 0 'base ''AulaData
 
+
 type AulaLens a = Lens' AulaData a
 type AulaGetter a = Getter AulaData a
 type AulaSetter a = Setter' AulaData a
 type AulaTraversal a = Traversal' AulaData a
 
+
+data DbField a where
+    DbSpaceSet :: DbField (Set IdeaSpace)
+    DbIdeaMap :: DbField Ideas  -- TODO: are these needed? e.g., for removal? or union of two maps?
+    DbIdea :: AUID Idea -> DbField Idea
+    DbUserMap :: DbField Users
+    DbUser :: AUID User -> DbField User
+    DbTopicMap :: DbField Topics
+    DbTopic :: AUID Topic -> DbField Topic
+    DbDelegationMap :: DbField Delegations
+    DbDelegation :: AUID Delegation -> DbField Delegation
+    DbElaborationDuration :: DbField DurationDays
+    DbVoteDuration :: DbField DurationDays
+    DbSchoolQuorum :: DbField Percent
+    DbClassQuorum :: DbField Percent
+    DbLastId :: DbField Integer
+    DbCompleteAulaData :: DbField AulaData
+-- TODO: add more like that, pity it's not composable; or is it?
```

Comment:
Actually, I meant more like `DbIdeaComment` below (my comments always refer to the item below, unless they start with a caret ^ or are indented wrt to the previous line). But yes, what you described is less constructors than the 10 more extra constructors like `DbIdeaComment` needed right now and more in the future.